session/tidb.go

```diff
@@ -53,6 +53,7 @@ func (dm *domainMap) Get(store kv.Storage) (d *domain.Domain, err error) {
 	// If this is the only domain instance, and the caller doesn't provide store.
 	if len(dm.domains) == 1 && store == nil {
 		for _, r := range dm.domains {
+			dm.mu.RUnlock()
```

Comment:
Or should we use `defer` in line52?

Reply:
it is an optimization that will unlock in advance， don't unlock when to return. I think the front part can be a function that uses ```defer``` to unlock.